### ISSUE TITLE
error : Failed to resolve SDK 'Microsoft.DotNet.Helix.Sdk'. Package restore was successful but a package with the ID of "Microsoft.DotNet.Helix.Sdk" was not installed.

### ISSUE THREAD

**Issue by elinor-fung** [dotnet/coreclr#26057]
PR and outerloop builds are intermittently hitting this when sending tests to Helix:
```
/__w/2/s/tests/helixpublishwitharcade.proj(7,3): error : Failed to resolve SDK 'Microsoft.DotNet.Helix.Sdk'. Package restore was successful but a package with the ID of "Microsoft.DotNet.Helix.Sdk" was not installed.
```
The strange thing is that it looks like the test run jobs are actually sent and completed after the error.

Examples:
- PR build [293716](https://dev.azure.com/dnceng/public/_build/results?buildId=293716&view=logs&lineStart=49&jobId=6c23f58b-dd4e-59d4-0149-12bcf7fa0bde&taskId=4150e432-679f-5945-da61-a2b1889dc85c)
- PR build [298122](https://dev.azure.com/dnceng/public/_build/results?buildId=298122&view=logs&lineStart=42&jobId=9c532748-9ca4-586a-2db0-c47bd86f8652&taskId=1c6bd9fc-3899-5882-0618-d192d93e914c)
- outerloop build [286673](https://dev.azure.com/dnceng/public/_build/results?buildId=286673&view=logs&lineStart=46&jobId=02288672-4313-54e7-ac78-0c78077b23f1&taskId=7d1b159a-1648-5a13-c85c-cd5b5d582719)
- outerloop build [296676](https://dev.azure.com/dnceng/public/_build/results?buildId=296676&view=logs&lineStart=50&jobId=5df70884-47c0-5c34-2344-7b08ed4395e9&taskId=f3456839-a8d5-574a-8533-4ebf43d189a8)

cc @dotnet/coreclr-infra

**Comment by hoyosjs** [comment]
This keeps hitting several PRs and is related to bug NuGet/Home#7817 fixed by NuGet/NuGet.Client#2734. The SDK that we use now has the fix nonetheless, so we either have a bug in NuGet or the error string still gets in the stream if the casing is different even though it can pick it up. Given all our packages are lower cased by default, a work around for this might be to set the SDKs to lower case in the global.json and the files that reference Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Helix.Sdk. This feels very brittle nontheless and I haven't been able to repro it under a debugger to be able to say if it's the right fix.

**Comment by hoyosjs** [comment]
Another instance in https://github.com/dotnet/coreclr/pull/27062, and looks like the suggested workaround doesn't fully work. One of the referenced issues seems to point the RCA was wrong and the bug is not fully understood yet.

**Comment by jkotas** [comment]
Another instance in https://github.com/dotnet/coreclr/pull/27314

**Comment by jkotas** [comment]
@dotnet/coreclr-infra

**Comment by jkotas** [comment]
Another instance in https://github.com/dotnet/coreclr/pull/27317

**Comment by jkotas** [comment]
Another instance in #https://github.com/dotnet/coreclr/pull/27315 . I will stop commenting on all affected PRs. This issue affected like half of the PRs from last night.

**Comment by AndyAyersMS** [comment]
Another instance in #27337

**Comment by MattGal** [comment]
another instance: https://dev.azure.com/dnceng/public/_build/results?buildId=399304&view=logs

**Comment by MattGal** [comment]
This matches https://github.com/NuGet/Home/issues/7817 precisely.  I'll follow up with Nuget folks and see if I can get some guidance.

**Comment by markwilkie** [comment]
Is NuGet engaged on this yet?

**Comment by hoyosjs** [comment]
The proposed solution didn't work, and there's been reports of that failing elsewhere. The cause was not fully understood last time I looped in. The only other suggested solution was restoring the SDK packages as a separate step before any build step. I believe other repos tried to solve this by adding this as a tool under Tools.props

**Comment by echesakovMSFT** [comment]
@hoyosjs Do you think it would help to restore the Sdk packages as a separate task inside helixpublishwitharcade.proj before going in parallel?

Something like this?
```xml
   <MSBuild Projects="$(MSBuildProjectFile)" Targets="PreparePayloadsDirectories" Properties="Scenarios=$(_Scenarios)" StopOnFirstFailure="true" />
 + <MSBuild Projects="$(MSBuildProjectFile)" Targets="Restore" Properties="UsesHelixSdk=true" />
```

**Comment by echesakovMSFT** [comment]
It could be due to a race when NuGet tries to restore the packages from multiple threads

**Comment by markwilkie** [comment]
Adding @riarenas and @JohnTortugo  as they'll want to follow along.

**Comment by JohnTortugo** [comment]
> This matches [NuGet/Home#7817](https://github.com/NuGet/Home/issues/7817) precisely. I'll follow up with Nuget folks and see if I can get some guidance.

This is an interesting bug. I wonder if we got any comments from the NuGet team yet. @MattGal @dtivel
